docker/Dockerfile

```diff
@@ -126,12 +126,13 @@ COPY --from=binaries \
     linera-db \
     ./
 
-COPY --chmod=755 \
-    docker/server-entrypoint.sh \
+COPY docker/server-entrypoint.sh \
     docker/server-init.sh \
     configuration/"$environment"/validator_1.toml \
     ./
 
+RUN chmod 755 server-entrypoint.sh server-init.sh
```

Comment:
What's happening here? I usually prefer to avoid creating the extra layer if we can — does Cloud Build's Docker not support `--chmod`?

Reply:
From the PR's description:

> Also, COPY --chmod requires BuildKit, so just doing separate commands as installing BuildKit on GCP seems non-trivial.

Reply:
Ah, I guess one thing for me to test is that the current version of Docker I’m using might have BuildKit, and this change won’t be needed anymore. I’ll do that before merging

Reply:
Thanks @christos-h, I guess my reading skills were not on top form this morning :sweat_smile: 

I think if we no longer always try to build `builder_copy` then that's an indication that we're using BuildKit — so this shouldn't be needed.